src/Microsoft.Azure.SignalR/ServerConnections/ServiceConnection.cs

```diff
@@ -66,15 +67,20 @@ protected override Task DisposeConnection()
             return _connectionFactory.DisposeAsync(connection);
         }
 
-        protected override async Task CleanupConnections()
+        protected override async Task CleanupConnections(string instanceId = null)
         {
             try
             {
                 if (_connectionIds.Count == 0)
                 {
                     return;
                 }
-                await Task.WhenAll(_connectionIds.Select(s => PerformDisconnectAsyncCore(s.Key, false)));
+                var connectionIds = _connectionIds.Select(s => s.Key);
+                if (!string.IsNullOrEmpty(instanceId))
+                {
+                    connectionIds = _connectionIds.Where(s => s.Value == instanceId).Select(s => s.Key);
+                }
```

Comment:
```
var connectionIds = _connectionIds;

if (!string.IsNullOrEmpty(instanceId)){
connectionIds = _connectionIds.Where(s=>s.Value==instanceId);
}

 await Task.WhenAll(connectionIds.Select(s => PerformDisconnectAsyncCore(s.Key, false)));
```

Reply:
Tried before, but `Where` filter will return `Generic Dictionary` and can not convert to `ConcurrentDictionary`.